lib/string.js

```diff
@@ -374,8 +201,70 @@ internals.String.prototype.email = function (isEmailOptions) {
 };
 
 
+internals.String.prototype.ip = function (ipOptions) {
+    var regex = internals.ipRegex;
+    var versions;
+    if (ipOptions) {
+        Hoek.assert(typeof ipOptions === 'object', 'ipOptions must be an object');
+
+        if (ipOptions.cidr) {
+            Hoek.assert(typeof ipOptions.cidr === 'string', 'ipOptions.cidr must be a string');
+            ipOptions.cidr = ipOptions.cidr.toLowerCase();
+
+            Hoek.assert(ipOptions.cidr in Ip.cidrs, 'ipOptions.cidr must be one of ' + Object.keys(Ip.cidrs).join(', '));
+
+            // If we only received a `cidr` setting, create a regex for it. But we don't need to create one if `cidr` is "optional" since that is the default
+            if (!ipOptions.version && ipOptions.cidr !== 'optional') {
+                regex = Ip.createIpRegex(['ipv4', 'ipv6', 'ipvfuture'], ipOptions.cidr);
+            }
+        } else {
+            // Set our default cidr strategy
+            ipOptions.cidr = 'optional';
+        }
+
+        if (ipOptions.version) {
+            if (!Array.isArray(ipOptions.version)) {
```

Comment:
You should check the length if it's an array as well, createIpRegex will be invalid otherwise.